### ISSUE TITLE
Is it ok to load ifc file incrementally?

### ISSUE THREAD

**Issue by allenwwg** [tomvandig/web-ifc#155]
Let's assume these steps:
1. load an IFC file into Web-IFC viewer
2. Some edit operations (maybe adding, deleting) by other application to the IFC file
3. load the IFC file again
4. question is: Is it possible just load the changes?

I looked into the code, an ifc file is loaded as a mesh model which contains geometries, is it ok do something to the mesh model to implement incremental loading?

**Comment by agviegas** [comment]
Hi @allenwwg,

No, sorry, right now web-ifc can only load full IFC files. As you know, what we usually call "IFC file" is a "STEP file", which is very inconvenient for this kind of things. 

I don't know what you are trying to build, but if you give us more details we'll try to give you some alternative. Right now we are implementing a new system that converts the STEP file to a JSON (properties) and GLB files (geometry). You can see the export example and the [import example](https://github.com/IFCjs/hello-world/tree/main/examples/web-ifc-viewer/gltf-import) and the [export example](https://github.com/IFCjs/hello-world/tree/main/examples/web-ifc-viewer/gltf-export). 

This way you can load BIM files incrementally and much faster. You can see it in action [here](https://ifcjs.github.io/hello-world/examples/web-ifc-viewer/gltf-import/index.html). We've adapted all IFC.js tools to work with this workflow as well, and you could also save the properties in a proper database.

In a nutshell, if you want to build a scalable and efficient BIM platform that is able to load changes fast, you shouldn't make it rely on STEP, in my opinion. Cheers!
